src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -421,7 +427,11 @@ private Transaction assembleTransaction(
       List<SorobanAuthorizationEntry> newEntries = new ArrayList<>(originalEntries);
       if (simulateHostFunctionResult.getAuth() != null) {
         for (String auth : simulateHostFunctionResult.getAuth()) {
-          newEntries.add(sorobanAuthorizationEntryFromXdrBase64(auth));
+          try {
+            newEntries.add(SorobanAuthorizationEntry.fromXdrBase64(auth));
```

Comment:
@overcat , please refer to the js soroban client for a reference on how it merges simulation results to original transaction,

https://github.com/stellar/js-soroban-client/blob/main/src/transaction.ts#L21

the java sdk should be consistent to how that handles merging auths, the js sdk will not apply simulated auths if the original tx had auths.

Reply:
It seems that this is a recent change introduced in the JS SDK. Let's track it in the [new issue](https://github.com/stellar/java-stellar-sdk/issues/504).